huxley/api/tests/test_committee_feedback.py

```diff
@@ -0,0 +1,124 @@
+# copyright (c) 2011-2015 Berkeley Model United Nations. All rights reserved.
+# Use of this source code is governed by a BSD License (see LICENSE).
+
+from django.core.exceptions import ValidationError
+
+from huxley.accounts.models import User
+from huxley.api import tests
+from huxley.api.tests import auto
+from huxley.utils.test import models
+import random
+
+
+class CommitteeFeedbackDetailCreateTestCase(tests.PostSingleAPITestCase):
+    url_name = 'api:committee_feedback_detail'
+
+    def setUp(self):
+        self.committee = models.new_committee(name='CYBER')
+        self.assignment = models.new_assignment(committee=self.committee)
+        self.delegate = models.new_delegate(assignment=self.assignment)
+        self.params = {
+            'comment': "I never got called on ever. SAD!",
+            'committee': self.committee.id
+        }
+
+    def test_anonymous_user(self):
+        '''Anonymous users cannot create feedback'''
+        response = self.get_response(self.committee.id, params=self.params)
+        self.assertNotAuthenticated(response)
+
+    def test_delegate(self):
+        '''Delegate can create feedback only once'''
+        self.user = models.new_user(
+            username='delegate',
+            password='delegate',
+            user_type=User.TYPE_DELEGATE,
+            delegate=self.delegate,
+            assignment=self.assignment)
+        self.client.login(username='delegate', password='delegate')
+        self.assertFalse(self.user.delegate.committee_feedback_submitted)
+        response_1 = self.get_response(self.committee.id, params=self.params)
+        self.assertEqual(response_1.data, {
+            "committee": self.committee.id,
+            "comment": self.params['comment'],
+        })
+
+        #This is how to refresh an object after it was updated from somewhere else
+        self.user.delegate.refresh_from_db()
+        self.assertTrue(self.user.delegate.committee_feedback_submitted)
+        response_2 = self.get_response(self.committee.id, params=self.params)
+        print(response_2)
+        self.assertFeedbackPreviouslySubmitted(response_2)
```

Comment:
What's going on here?

Reply:
This test checks that a delegate can only submit one feedback.

Because we don't store the delegate_id anywhere on the CommitteeFeedback model to maintain anonymous feedback, the only real way to check to see if a delegate has submitted feedback previously is the committee_feedback_submitted field on the Delegate model. This is a check to see that this is being updated properly.

Reply:
We could separate it into separate tests if you think that's necessary? But I don't think it is